ouroboros-consensus/src/Ouroboros/Consensus/TxSubmission.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE FlexibleContexts    #-}
+
+module Ouroboros.Consensus.TxSubmission
+  ( localTxSubmissionServer
+  ) where
+
+import           Control.Tracer
+
+import           Ouroboros.Network.Protocol.LocalTxSubmission.Server
+
+import           Ouroboros.Consensus.Mempool.API
+
+
+-- | Local transaction submission server, for adding txs to the 'Mempool'
+--
+localTxSubmissionServer
+  :: (Monad m, Show (ApplyTxErr blk)) -- TODO: consider using condense
+  => Tracer m unused --TODO: trace adding txs here or in the mempool directly
+  -> Mempool m blk
+  -> LocalTxSubmissionServer (GenTx blk) String m ()
+localTxSubmissionServer _tracer Mempool{addTxs} =
+    server
+  where
+    server = LocalTxSubmissionServer {
+      recvMsgSubmitTx = \tx -> do
+        res <- addTxs [tx]
```

Comment:
The current demo is showing the hash of accepted transactions, which is useful for submitting a following tx which spends output of it. Is this covered by `addTxs`?

Reply:
I don't think it is. We should indeed trace the txs being accepted.